main.cpp

```diff
@@ -35,74 +35,66 @@ struct T
 
 struct X                                //4
 {
-    T* compare(T* a, T* b) //5
+    const T* compare(const T & a, const T & b) //5
     {   
-        if( a != nullptr && b != nullptr )
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
+        if( a.value < b.value ) return &a;
+        else if( a.value > b.value ) return &b;
+
         return nullptr;
     }
 };
 
 struct U
 {
     float first { 0 }, second { 0 };
-    float memberUpdate(float* newVal)      //12
+    float memberUpdate(const float & newVal)      //12
```

Comment:
normally, primitives are passed by value, not by reference.  This is fine for the sake of this project.